src/Components/ListOfGoods/ListOfGoods.jsx

```diff
@@ -0,0 +1,131 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export class ListOfGoods extends React.Component {
+  state = {
+    reverse: false,
+    sortBy: '',
+    selectLength: 1,
+  };
+
+  reset() {
+    this.setState({
+      reverse: false,
+      sortBy: '',
+      selectLength: 1,
+    });
+  }
+
+  goodsRevers() {
+    this.setState(state => ({
+      reverse: !state.reverse,
+    }));
+  }
+
+  goodsSortByLength() {
+    this.setState({ sortBy: 'length' });
+  }
+
+  goodsSortByName() {
+    this.setState({ sortBy: 'name' });
+  }
+
+  selectLength(event) {
+    this.setState({ selectLength: event.target.value });
+  }
+
+  render() {
+    const { goods } = this.props;
+    const { selectLength, sortBy, reverse } = this.state;
+
+    const numbers = [...Array(10)].map((n, i) => i + 1);
```

Comment:
create an array outside of the component, no need to render new one for every render